uvloop/loop.pyx

```diff
@@ -1674,25 +1635,68 @@ cdef class Loop:
                     while addrinfo != NULL:
                         if addrinfo.ai_family == uv.AF_UNSPEC:
                             raise RuntimeError('AF_UNSPEC in DNS results')
 
-                        tcp = self._create_server(
-                            addrinfo.ai_addr, protocol_factory, server,
-                            ssl, reuse_port, backlog,
-                            ssl_handshake_timeout, ssl_shutdown_timeout)
+                        try:
+                            sock = socket_socket(addrinfo.ai_family,
+                                                 addrinfo.ai_socktype,
+                                                 addrinfo.ai_protocol)
+                        except socket_error:
+                            # Assume it's a bad family/type/protocol combination.
+                            if self._debug:
+                                aio_logger.warning('create_server() failed to create '
+                                                   'socket.socket(%r, %r, %r)',
+                                                    addrinfo.ai_family,
+                                                    addrinfo.ai_socktype,
+                                                    addrinfo.ai_protocol, exc_info=True)
+                            continue
 
-                        server._add_server(<TCPServer>tcp)
+                        if reuse_address:
+                            sock.setsockopt(uv.SOL_SOCKET, uv.SO_REUSEADDR, 1)
+                        if reuse_port:
+                            sock.setsockopt(uv.SOL_SOCKET, uv.SO_REUSEPORT, 1)
+                        # Disable IPv4/IPv6 dual stack support (enabled by
+                        # default on Linux) which makes a single socket
+                        # listen on both address families.
+                        if addrinfo.ai_family == uv.AF_INET6:
+                            sock.setsockopt(uv.IPPROTO_IPV6, IPV6_V6ONLY, 1)
+
+                        pyaddr = __convert_sockaddr_to_pyaddr(addrinfo.ai_addr)
+                        try:
+                            sock.bind(pyaddr)
+                        except OSError as err:
+                            raise OSError(err.errno, 'error while attempting '
+                                          'to bind on address %r: %s'
+                                          % (pyaddr, err.strerror.lower())) from None
+
+                        tcp = TCPServer.new(self, protocol_factory, server,
+                                            uv.AF_UNSPEC, backlog,
+                                            ssl, ssl_handshake_timeout,
+                                            ssl_shutdown_timeout)
+
+                        try:
+                            tcp._open(sock.fileno())
+                        except Exception:
+                            tcp._close()
+                            raise
+
+                        tcp._attach_fileobj(sock)
```

Comment:
I think we can try calling `sock.detach()` and skip this line.  There's no point in keeping the reference to the original socket here.  Hopefully that will fix the warning you saw.

Reply:
OK I'll do a `sock.detach()` followed by `sock = None`.

Regarding the warning I saw, it was something else: if any calls raise an exception we need to ensure the socket gets closed. That is why I keep track of `sock` and close it in the `finally`, otherwise we need try/catch every function call and do sock.close() at every possible failure.